charts/prometheus-thanos/templates/bucket-web-deployment.yaml

```diff
@@ -0,0 +1,74 @@
+{{- if .Values.bucketweb.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "prometheus-thanos.fullname" . }}-bucketweb
+  labels:
+    app.kubernetes.io/name: {{ include "prometheus-thanos.name" . }}-bucketweb
+    helm.sh/chart: {{ include "prometheus-thanos.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.bucketweb.replicaCount }}
+  strategy:
+  {{- with .Values.bucketweb.updateStrategy }}
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "prometheus-thanos.name" . }}-bucketweb
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "prometheus-thanos.name" . }}-bucketweb
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        prometheus-thanos-peer: "true"
+        {{- with .Values.bucketweb.additionalLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    {{- if .Values.bucketweb.additionalAnnotations }}
+      annotations:
+        {{- with .Values.bucketweb.additionalAnnotations }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    {{- end }}
+    spec:
+      containers:
+        - name: {{ .Chart.Name }}-bucketweb
+          imagePullPolicy: {{ .Values.bucketweb.image.pullPolicy }}
+          image: "{{ .Values.bucketweb.image.repository }}:{{ .Values.bucketweb.image.tag }}"
+          args:
```

Comment:
To specify GCP credentials, we need the possibility of defining arbitrary environment variables, volumes and volumeMounts. See storegatway for example.
Could you please add it?

Reply:
Thanks for the feedback, and good catch, I broke response to the feedback out into 2 commits as the rebase was a minor pain due to my own changes to the Readme.